.travis.yml

```diff
@@ -67,14 +67,6 @@ jobs:
       - brew update && brew unlink python@2 && brew install gpgme
       script:
       - hack/travis_osx.sh
-    - stage: local-build
-      <<: *local-build
-      name: local build for linux
-      os: linux
-      services:
-      - docker
-      script: 
-      - make vendor && ./hack/tree_status.sh && make local-cross && make check
```

Comment:
Is there anything providing the equivalent of `make vendor && ./hack/tree_status.sh` (i.e. ensure that we are not committing any patches, or missing any updates, to the `vendor` subdirectory?

Reply:
Yes (I just checked).  This should be [caught by the `validate` task](https://github.com/containers/skopeo/blob/main/.cirrus.yml#L64) (unless I'm missing something).

Reply:
`validate-local` AFAICS does not do this; there isn’t a single reference to `hack/tree_status.sh`, and I can’t find any equivalent in the repo.

(The exclusion from `validate-local` is somewhat intentional or at least useful: It’s useful to run `make validate-local` during development _on uncommitted code_, and that would trigger `tree_status.sh`.

The larger picture is that _Skopeo_ PRs, we should never need to run `make vendor` _on top_; the PR should already contain an update to the `vendor` subdirectory. So `make vendor && ./hack/tree_status.sh` is a sanity check that the PR is self-contained and no further `make vendor` is necessary. OTOH if c/image tests replace the c/image implementation vendored into Skopeo, we do need to run `make vendor` _somewhere_, and _not_ fail c/image tests on uncommitted `vendor/containers/image`.)

So maybe we want to:
- Do this “`vendor` is fully committed” check in `validate_task`
- I don’t really care where the code lives, a `make validate-vendor-is-committed` in the `Makefile` (along with comments about why it is not included in `validate-local`), keeping `hack/tree_status.sh`, would probably be most consistent with the rest; alternatively all in `Makefile` and getting rid of the separate shell script, or all only in `.cirrus.yml` and nothing in `Makefile`, would be fine with me as well.

Reply:
> So maybe we want to:

That sounds good.  IIRC, c/image has a similar 'validate' task...crap, it seems that doesn't have a `make vendor` +  `hack/tree_status.sh` either :disappointed: 

Yes we should keep `hack/tree_status.sh` for consistency.

I just looked...podman does it in a shell script (not Makefile), buildah does it directly in `.cirrus.yml`. 

Okay, I opened a PR: https://github.com/containers/skopeo/pull/1340

Reply:
c/image doesn’t need a `tree_status.sh` because it doesn’t maintain the `vendor` subdirectory as part of the repo. (Vendoring dependencies doesn’t work well for libraries that have vendored types as part of the API.)

Reply:
Ya, I noticed that too (after I actually went looking for it).  Okay, we should be all caught up now then (I think).  I's dotted and T's crossed :smile: Let me know if you spot any other blunders I made.

Reply:
> Okay, we should be all caught up now then (I think). I's dotted and T's crossed 😄 

Yup. Thanks again!